src/app/frank/fd_frank_verify.c

```diff
@@ -171,15 +264,158 @@ run( fd_frank_args_t * args ) {
+       mline at time now.  Speculatively processs it here. */
+    ulong vin_data_sz    = (ulong)vin_mline->sz;
+    uchar *  vin_dcache_chunk_laddr = (uchar *)fd_chunk_to_laddr( vin_wksp, vin_mline->chunk );
+    uchar *  udp_payload = (uchar *)fd_chunk_to_laddr( wksp, chunk );
+    memcpy(udp_payload, vin_dcache_chunk_laddr, vin_data_sz);
+
+    /* Check that we weren't overrun while processing */
+    vin_seq_found = fd_frag_meta_seq_query( vin_mline );
+    if( FD_UNLIKELY( fd_seq_ne( vin_seq_found, vin_mcache_seq ) ) ) {
+      vin_accum_ovrnr_cnt++;
+      FD_LOG_INFO(( "verifyin.%lu ovrnr encountered:   vin_mcache_seq=%lu   vin_seq_found=%lu   vin_accum_ovrnr_cnt=%lu", args->tile_idx, vin_mcache_seq, vin_seq_found, vin_accum_ovrnr_cnt ));
+      vin_mcache_seq = vin_seq_found;
+      now = fd_tickcount();
+      continue;
+    }
+
+    vin_accum_pub_cnt++;
+    vin_accum_pub_sz += vin_data_sz;
+
+    /* Wind up for the next iteration for verifyin */
+    vin_mcache_seq   = fd_seq_inc( vin_mcache_seq, 1UL );
+    vin_mline = vin_mcache + fd_mcache_line_idx( vin_mcache_seq, vin_mcache_depth );
+
+    ushort payload_sz = *(ushort*) (udp_payload + vin_data_sz - sizeof(ushort));
+    fd_txn_t * txn = (fd_txn_t*) fd_ulong_align_up( (ulong)(udp_payload) + payload_sz, 2UL );
+
+    ulong const * public_key = (ulong const *)(udp_payload + txn->acct_addr_off);
+    ulong const * sig        = (ulong const *)(udp_payload + txn->signature_off);
+    uchar const * msg        = (uchar const *)(udp_payload + txn->message_off);
+    ulong msg_sz             = (ulong)payload_sz - txn->message_off;
+
+    /* Sig is already effectively a cryptographically secure hash of
+        public_key/private_key and message and sz.  So use this to do a
+        quick dedup of ha traffic (FIXME: POTENTIAL DOS ATTACK IF
+        SOMEBODY COULD INTERCEPT TRAFFIC AND SUBMIT PACKETS WITH SAME
+        PUBLIC KEY, SIG AND GARBAGE MESSAGE AHEAD OF THE TRAFFIC ...
+        SEEMS UNLKELY AS THEY WOULD EITHER BE BEHIND THE INBOUND OR BE
+        A MITM THAT COULD JUST DISCARD INBOUND TRAFFIC).
+
+        When running synthetic load though, we only have a very limited
+        set of messages and this dedup will be overly aggressive (as it
+        will spuriously matching earlier synthetic packets since they
+        are not resigned continuously)  So we just mock this up for the
+        time being. */
+
+    ulong ha_tag = *sig;
+    int ha_dup;
+    FD_TCACHE_INSERT( ha_dup, tcache_oldest, _tcache_ring, tcache_depth, _tcache_map, tcache_map_cnt, ha_tag );
+    if( FD_UNLIKELY( ha_dup ) ) { /* optimize for the non dup case */
+      accum_ha_filt_cnt++;
+      accum_ha_filt_sz += payload_sz; //WW accum_ha_filt_sz += msg_framing + msg_sz;
+      now = fd_tickcount();
+      continue;
+    }
+
+    /* We appear to have a message to verify.  So verify it.
+
+       When running synthetic load, the synthetic data will not fail
```

Comment:
Is this comment true on main?

Reply:
I'm not sure - for this PR I'm focusing on getting the wiring correct so packets flow from quic -> verify -> dedup.

In the next PR I'll make sure we are correctly checking signatures and not artificially inducing verification errors.